src/protocol/mod.rs

```diff
@@ -449,9 +457,14 @@ impl WebSocketContext {
                     } else {
                         // The server MUST close the connection upon receiving a
                         // frame that is not masked. (RFC 6455)
-                        return Err(Error::Protocol(
-                            "Received an unmasked frame from client".into(),
-                        ));
+                        // The only exception here is if the user explicitly accepts given
+                        // stream (by tungstenite::server::accept_with_config or tungstenite::server::accept_hdr_with_config)
+                        // with WebSocketConfig.server_allow_unmasked set to true
+                        if self.get_config().server_allow_unmasked == false {
```

Comment:
We do not need to use `get_config()` as `config` is the field of the `self`.

Additionally `self.server_allow_unmasked == false` is the same as `!self.server_allow_unmasked`.